manager/controllers/app/m4dapplication_controller.go

```diff
@@ -381,3 +390,30 @@ func (r *M4DApplicationReconciler) GetAllModules() (map[string]*app.M4DModule, e
 	}
 	return moduleMap, nil
 }
+
+// GetProcessingGeography determines the geography of the workload cluster.
+// If no workload has been specified, a local cluster is assumed.
+func (r *M4DApplicationReconciler) GetProcessingGeography(applicationContext *app.M4DApplication) (string, error) {
+	clusterName := applicationContext.Spec.Selector.ClusterName
+	if clusterName == "" {
+		clusterMetadataConfigmap := corev1.ConfigMap{}
+		namespacedName := client.ObjectKey{
+			Name:      "cluster-metadata",
+			Namespace: utils.GetSystemNamespace(),
+		}
+		if err := r.Client.Get(context.Background(), namespacedName, &clusterMetadataConfigmap); err != nil {
+			return "", err
+		}
+		return clusterMetadataConfigmap.Data["Region"], nil
```

Comment:
That's an implementation detail of `local.ClusterManager`. If we need an indication of the local cluster or an indication of the primary cluster then let's add it to the cluster manager interface instead.

Reply:
Added a method GetLocalCluster to the ClusterLister interface.

razee implementation of this function is not yet done - I need help with this. Local implementation uses GetClient() of manager which is not passed to razee manager. Either to pass this argument, or is there a "razee native" way to use the local config?

Reply:
@froesef ^

Reply:
Using local implementation to get local cluster information